src/btf_parser.cpp

```diff
@@ -98,27 +111,713 @@ void btf_parse_line_information(const std::vector<uint8_t>& btf, const std::vect
+            kind_float.size_in_bytes = btf_type->size;
+            kind = kind_float;
+            break;
+        }
+        case BTF_KIND_DECL_TAG: {
+            btf_kind_decl_tag kind_decl_tag;
+            if (remaining_size < sizeof(btf_decl_tag_t)) {
+                throw std::runtime_error("Invalid .BTF section - invalid BTF_KIND_DECL_TAG size");
+            }
+            auto btf_decl_tag = reinterpret_cast<const btf_decl_tag_t*>(btf.data() + type_offset + sizeof(btf_type_t));
+            // Name is required.
+            kind_decl_tag.name = _btf_find_string(string_table, btf_type->name_off);
+            kind_decl_tag.type = btf_type->type;
+            kind_decl_tag.component_index = btf_decl_tag->component_idx;
+            kind = kind_decl_tag;
+            size += sizeof(btf_decl_tag_t);
+            break;
+        }
+        case BTF_KIND_TYPE_TAG: {
+            btf_kind_type_tag kind_type_tag;
+            // Name is required.
+            kind_type_tag.name = _btf_find_string(string_table, btf_type->name_off);
+            kind_type_tag.type = btf_type->type;
+            kind = kind_type_tag;
+            break;
+        }
+        case BTF_KIND_ENUM64: {
+            uint32_t enum_count = BPF_TYPE_INFO_VLEN(btf_type->info);
+            uint32_t required_size = enum_count * sizeof(btf_enum64_t);
+            // Check for overflow.
+            if (required_size < enum_count) {
+                throw std::runtime_error("Invalid .BTF section - invalid BTF_KIND_ENUM size");
+            }
+            if (remaining_size < required_size) {
+                throw std::runtime_error("Invalid .BTF section - invalid BTF_KIND_ENUM size");
+            }
+            btf_kind_enum64 kind_enum;
+            for (uint32_t index = 0; index < enum_count; index++) {
+                auto btf_enum = reinterpret_cast<const btf_enum64_t*>(btf.data() + type_offset + sizeof(btf_type_t) +
+                                                                      index * sizeof(btf_enum_t));
+                btf_kind_enum64_member member;
+                member.name = _btf_find_string(string_table, btf_enum->name_off);
+                member.value = (static_cast<uint64_t>(btf_enum->val_hi32) << 32) | btf_enum->val_lo32;
+                kind_enum.members.push_back(member);
+            }
+            kind_enum.is_signed = BPF_TYPE_INFO_KIND_FLAG(btf_type->info);
+            // Name is optional.
+            if (btf_type->name_off) {
+                kind_enum.name = _btf_find_string(string_table, btf_type->name_off);
+            }
+            kind_enum.size_in_bytes = btf_type->size;
+            kind = kind_enum;
+            size += required_size;
+            break;
+        }
+        default: throw std::runtime_error("Invalid .BTF section - invalid BTF_KIND");
+        }
+        visitor(++id, kind);
+        type_offset += size;
+
+    }
+}
+
+std::string pretty_print_json(const std::string& input)
+{
```

Comment:
```suggestion
std::string pretty_print_json(const std::string& input) {
```
Similarly everywhere else.